pkg/database/openshiftclusters.go

```diff
@@ -293,3 +294,19 @@ func (c *openShiftClusters) partitionKey(key string) (string, error) {
 	r, err := azure.ParseResourceID(key)
 	return r.SubscriptionID, err
 }
+
+func (c *openShiftClusters) ValidateUniqueKey(ctx context.Context, partitionKey, uniqueKeyName, uniqueKeyValue string) (*api.OpenShiftClusterDocuments, error) {
```

Comment:
@jim-minter  let me know if this is better to have 2 seperated function inside `openshiftclusters.go` one per unique key, or if this is ok like this ?

Reply:
Two separate functions is preferred.  `" + uniqueKeyName + "` is uncomfortable because it's one of the two prerequisites for a SQL injection attack down the road.  If you do this, you should revalidate in this function that uniqueKeyName is a known value.  Probably two functions `GetByClientID` and `GetByClusterResourceGroupID` would be better.